bricks/next-builder/src/search-tree/utils.tsx

```diff
@@ -0,0 +1,338 @@
+  meta: {
+    color: "cyan",
+    icon: "cogs",
+    lib: "fa",
+    prefix: "fas",
+  },
+};
+
+const ingoreKey = ["bricks", "slots"];
+
+const supportKey = [
+  "routes",
+  "bricks",
+  "slots",
+  "meta",
+  "customTemplates",
+  "menus",
+  "i18n",
+  "app",
+  "zh",
+  "en",
+  "main",
+];
+
+let isParentRouteLock = false;
+
+export interface PlainObject extends Object {
+  [key: string]: any;
+}
+interface filterOption {
+  allowKeySearch?: boolean;
+}
+interface builTreeOptions {
+  parentPath?: string;
+  parentId?: string;
+  isSlots?: boolean;
+}
+
+function getType(item: PlainObject): keyof typeof iconTypeConstants {
+  if (item.brick) return "brick";
+
+  return item?.type;
+}
+
+function getTypeIcon(type: keyof typeof iconTypeConstants) {
+  if (iconTypeConstants[type]) {
+    return (
+      <GeneralIcon
+        style={{
+          fontSize: 12,
+          margin: "0 2px",
+        }}
+        icon={iconTypeConstants[type] as MenuIcon}
+      />
+    );
+  } else {
+    return (
+      <span
+        style={{
+          display: "inline-block",
+          width: 12,
+          height: 12,
+          margin: "0 2px",
+        }}
+      />
+    );
+  }
+}
+
+export function clone(obj: PlainObject) {
+  const newObj: PlainObject = Object.create(null);
+  for (const [k, v] of Object.entries(obj)) {
+    if (!ingoreKey.includes(k)) {
+      newObj[k] = v;
+    }
+  }
+  Object.getOwnPropertySymbols(obj).forEach((symbol) => {
+    newObj[symbol as any] = obj[symbol as any];
+  });
+  return newObj;
+}
+
+export function getTitle(item: PlainObject, defaultKey?: string): string {
+  return (
+    item?.menuId ||
+    item?.brick ||
+    item?.alias ||
+    item?.path ||
+    item?.name ||
+    item?.type ||
+    item?.title ||
+    defaultKey
+  );
+}
+
+function getPath(parentPath = "", childPath: string) {
+  return parentPath ? `${parentPath}/${childPath}` : childPath;
+}
+
+function traversalArray(
+  treeData: Array<PlainObject>,
+  options: builTreeOptions
+) {
+  const tree: Array<any> = [];
+  for (let i = 0; i < treeData.length; i++) {
+    const title = getTitle(treeData[i]);
+    const path = getPath(options?.parentPath ?? "", String(i));
+    let parentId = options.parentId;
+    if (!parentId && treeData[i][symbolForNodeId as any]) {
+      parentId = treeData[i][symbolForNodeId as any];
+    }
+    const child: PlainObject = {
+      title,
+      key: path,
+      icon: getTypeIcon(getType(treeData[i])),
+      [NODE_INFO]: clone({
+        ...treeData[i],
+        realParentId: parentId,
+      }),
+    };
+    const subTree = traversalData(treeData[i], {
+      ...options,
+      parentPath: path,
+      parentId,
+    });
+    if (subTree.length > 0) {
+      child.children = subTree;
+    }
+    tree.push(child);
+  }
+
+  return tree;
+}
+
+function traversalObject(treeData: PlainObject, options: builTreeOptions) {
+  const tree: Array<PlainObject> = [];
+  let isSlots = false;
+
+  for (const key in treeData) {
```

Comment:
`for...in` 会遍历原型链，这里建议：

```suggestion
  for (const key of Object.keys(treeData)) {
```

Reply:
可以